File: Freedom/大众点评/闪惠团购/DZDealController.m
//  DZDealController.m
//  Freedom
//  Created by Super on 15/12/1.
#import "DZDealController.h"
#import "DZDealListViewController.h"
@interface DZDealController (){
    BaseScrollView *contentScrollView;
}
@end
@implementation DZDealController
- (void)viewDidLoad {
    [super viewDidLoad];
    [self buildUI];
}
#pragma mark 搭建UI界面
-(void)buildUI{
    self.view.backgroundColor = [UIColor whiteColor];
    UIBarButtonItem *map = [[UIBarButtonItem alloc]initWithTitle:@"北京" style:UIBarButtonItemStylePlain actionBlick:^{
    }];
    self.navigationItem.leftBarButtonItem = map;
    self.navigationItem.title = @"团购";
    NSArray *titles = @[@"精选",@"享美食",@"点外卖",@"看电影",@"趣休闲"];
    NSArray *controllers = @[@"DZDealListViewController",@"DZDealListViewController",@"DZDealListViewController",@"DZDealListViewController",@"DZDealListViewController"];
    contentScrollView = [BaseScrollView sharedContentTitleViewWithFrame:CGRectMake(0, 0, APPW, APPH-100) titles:titles controllers:controllers inViewController:self];
    contentScrollView.selectBlock = ^(NSInteger index, NSDictionary *dict) {
        DLog(@"点击了%ld,%@",index,dict);
    };
}
@end

File: Freedom/大众点评/闪惠团购/DZDealController.swift
//
//  DZDealController.swift
//  Freedom
import UIKit
import BaseFile
import XExtension
class DZDealController: DZBaseViewController {
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    func buildUI() {
        view.backgroundColor = UIColor.white
        let map = UIBarButtonItem(title: "北京", style: .plain, actionBlick: {() -> Void in
        })
        navigationItem.leftBarButtonItem = map
        navigationItem.title = "团购"
        let titles = ["精选", "享美食", "点外卖", "看电影", "趣休闲"]
        let vc = DZDealListViewController()
        let controllers = [vc,vc,vc,vc,vc]
        let contentScrollView = BaseScrollView(contentTitleView: CGRect(x: 0, y: 0, width: APPW, height: APPH - 100), titles: titles, controllers:controllers, in: self)
        contentScrollView.selectBlock = {(_ index: Int, _ dict: [AnyHashable: Any]?) -> Void in
            Dlog("点击了\(String(describing: dict))")
        }
    }
}
